Week 11: Happy thoughts

monkeybusiness101 opened the pull request:
Link to my deployed project:

https://sleepy-carson-4649f3.netlify.app

My reflections on how this week's project turned out:

Fun but challenging project. I wished I knew that there were more videos to watch (Which Poya told us on Friday..)

Things I'd like to have clarified or explained in more detail:

IngelaL commented on code/src/Like.js:
```diff
@@ -0,0 +1,18 @@
+    }
+
+  return  <button onClick={() => {handleLike(_id)}} className="like-button">
+            <span role="img" aria-label="Heart emoji"> ❤️ </span>
```
Here you could add and if statement that if the button gets 1 or more clicks, it should change background color. Like this: style={{ background: hearts > 0 ? '#feacac': 'f2f0f0'}} (Then you also have to add the hearts prop in the component). On your web page the background color of the heart only changes temporarily (when clicking on it) But then it changes back to the original background color.

IngelaL commented on code/src/index.css:
```diff
@@ -1,13 +1,89 @@
+}
+
+.like-button:focus {
+  background-color: #ffadad;
```
This only make the color of the like button change color when clicking on it but when you click utside of it it goes back to the background color it had before. Once the happy thought get a like, the color should change permanantly. See comment in LikeComponent line 15.

IngelaL commented on code/src/index.css:
```diff
@@ -1,13 +1,89 @@
+  margin-top: 50px;
+  font-size: 12px;
+  padding: 5px;
+}
```
Here you could add a 'max-width' of maybe 400px so the boxes don't get so stretched out on a big screen. Same with the form.

IngelaL commented on code/src/Fetch.js:
```diff
@@ -0,0 +1,50 @@
+import moment from 'moment'
+
+import { Like } from 'Like'
+
```
Great that you separate the react import from the components import.

IngelaL commented on code/src/Fetch.js:
```diff
@@ -0,0 +1,50 @@
+            console.log('Request successful', text)
+        })
+        .catch((error) => {
+            console.error('Request failed', error)
```
👍

IngelaL reviewed (changes requested):
Hi Max! You have a great structure of your components and your code looks great in general. What you need to do is to fix two things with the heart like-button: 1. to make the background stay red when clicked. 2. To make the "heart-count" work. When adding a like it does'nt update the number (x 1) on the page. (However, it works "behind the scenes"). I also have some suggestions on making your code more readable in some places.

monkeybusiness101 commented:
@IngelaL Thank you so much for the feedback, that was awesome!
- I have removed unnecessary blank spaces and fixed some indentation
- I fixed so that the like button keeps it color if there are more than one like (thank you for that)
- I added so that like count is more dynamic

Let me know if you want me to have a look at anything else.

Great PR!

monkeybusiness101 commented:
I made it look nicer on desktop as well (thank you for the input on that)

IngelaL commented:
Looks great now Max!

EmOlsz commented on code/src/App.js:
```diff
@@ -1,9 +1,13 @@
     <div>
-      Find me in src/app.js!
+      <Create />
+      <Fetch />
```
Agree!

monkeybusiness101 replied:
Fixed

EmOlsz commented on code/src/Create.js:
```diff
@@ -0,0 +1,36 @@
+    const handleSubmit = event => {
+        
+        event.preventDefault()
+        console.log(text)
```
Let's delete all of those console.logs

monkeybusiness101 replied:
Fixed

EmOlsz commented on code/src/Create.js:
```diff
@@ -0,0 +1,36 @@
+
+export const Create = () => {
+    
+    const [text, addMessage] = useState("")
```
Names of state and state handlers should have same base so it's easy to detect their pairs. In your case, correct naming would be :

```
const [text, setText] = useState("");
```

or 

```
const [message, setMessage] = useState("");
```

monkeybusiness101 replied:
Fixed

monkeybusiness101 commented:
@EmOlsz Hi Maks! I have removed console.logs, changed the components to reflect their purpose better, and changed the usestate to correlate.

/Max
